### ISSUE TITLE
Incorrect error message when accessing protected member in a nested type

### ISSUE THREAD

**Issue by svick** [dotnet/roslyn#50960]
**Version Used**: VS 2019 version 16.8.2, master (d0d9a59) on SharpLab

**Steps to Reproduce**:

[Compile code like the following:](https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxAN0QSwGYE8A+AAgEwAEAQtHALABQA3nac6QA4ID28AxvACalCAFlIBZABQBKUvQC+debTolSAESwYBIClTqNaLQWQBycWHD5MW+w4eFiImAHbiOwAFZxepDpKu2ZfwCWcXF1BExNSV8AOglJAG4g4NJQyig4aMk4qSSDYMVDRVkgA)

```c#
class Base
{
    protected void M() {}
}

class Derived : Base
{
    class Nested
    {
        void Main(object o)
        {
            ((Derived)o).M();
            ((Base)o).M();
        }
    }
}
```

**Actual Behavior**:

The second call to `M` produces the following error:

> error CS1540: Cannot access protected member 'Base.M()' via a qualifier of type 'Base'; the qualifier must be of type 'Derived.Nested' (or derived from it)

**Expected Behavior**:

The second call to `M` produces the following error:

> error CS1540: Cannot access protected member 'Base.M()' via a qualifier of type 'Base'; the qualifier must be of type 'Derived' (or derived from it)

The type `Derived.Nested` is not part of the inheritance hierarchy, so it doesn't make sense to mention it here, the type that should be mentioned is `Derived`.

Also, I think it would be better if the error message didn't use the term "qualifier", since I think it's not going to be familiar for many developers.
